File: LongestConsecutiveSequesnce/LongestConsecutiveSequesnce.js
function LongestConsecutiveSequesnce(arr) {
  let S = new Set();
  let ans = 0;
  for (let i = 0; i < arr.length; i++) S.add(arr[i]);
  console.log(S);
  for (let i = 0; i < arr.length; i++) {
    if (!S.has(arr[i] - 1)) {
      let j = arr[i];
      while (S.has(j)) j++;
      ans = Math.max(ans, j - arr[i]);
    }
  }
  console.log(ans);
}
LongestConsecutiveSequesnce([1, 9, 3, 10, 4, 20, 2]);

File: LongestConsecutiveSequesnce/LongestConsecutiveSequesnce.py
def LongestConsecutiveSequesnce(arr):
    ans = 0
    s = set()
    for ele in arr:
        s.add(ele)
    for i in range(0, len(s), 1):
        if (arr[i]-1) not in s:
            j = arr[i]
            while(j in s):
                j += 1
                print(j-arr[i])
                ans = max(ans, j-arr[i])
    print(ans)


LongestConsecutiveSequesnce([1, 9, 3, 10, 4, 20, 2])
